festim/hydrogen_transport_problem.py

```diff
@@ -324,8 +325,9 @@ def define_temperature(self):
     def initialise_exports(self):
         """Defines the export writers of the model, if field is given as
         a string, find species object in self.species"""
-
         for export in self.exports:
+            export.t = []
+            export.data = []
```

Comment:
Doing it here is not the right place.
This iterates through all the exports, which can be derived quantities, but also VTXExport or XDMFExport etc.

Instead, we could do it in `write()`, making use of the `self._is_first_time_export` attribute. For example here:

https://github.com/festim-dev/FESTIM/blob/f08f0b5c83d11b511a129316d040b61d0bf4f3b1/festim/exports/surface_quantity.py#L75-L76

Reply:
Alternatively we could do it here, but with an additional type check to only do this for derived quantities.